src/main/scala/akka/contrib/process/BlockingProcess.scala

```diff
@@ -107,16 +122,20 @@ class BlockingProcess(
     }
 
     try {
-      val stdin =
-        context.actorOf(OutputStreamSubscriber.props(process.getOutputStream), "stdin")
+      val selfRef = context.self
 
-      val stdout =
-        context.watch(context.actorOf(InputStreamPublisher.props(process.getInputStream, stdioTimeout), "stdout"))
+      val stdin = Sink.outputStream(process.getOutputStream)
+        .mapMaterializedValue(_ andThen { case _ => selfRef ! StdinTerminated })
 
-      val stderr =
-        context.watch(context.actorOf(InputStreamPublisher.props(process.getErrorStream, stdioTimeout), "stderr"))
+      val stdout = Source.inputStream(process.getInputStream)
+        .withAttributes(Attributes.inputBuffer(1, 1))
```

Comment:
On merging this PR, we should create an issue for the Akka team to use these as more sensible defaults. You cannot tell how an input stream will behave i.e. how many elements it will deliver.